src/db.ts

```diff
@@ -426,10 +424,7 @@ export class Db {
     indexSpec: IndexSpecification,
     options?: CreateIndexesOptions
   ): Promise<string> {
-    return executeOperation(
-      this.client,
-      new CreateIndexOperation(this, name, indexSpec, resolveOptions(this, options))
-    );
+    return this.collection(name).createIndex(indexSpec, options);
```

Comment:
Why the throwaway collection vs just exec-ing your own createIndexOp?

Reply:
Using public APIs feels cleaner, and has the added benefit of a bit of extra testing cause we're using a complete public API, but I don't feel strongly.

For `indexInformation`, `IndexInformationOperation` no longer exists so I call `Collection.indexInformation` to share code

Reply:
We should probably determine some consistency around using our own public API or circumventing it, I am not sure what the right call is or if there is one that applies universally.

I see this the opposite way of clean because we could call `CreateIndexesOperation.fromIndexSpecification` with the name and indexSpec given, and resolveOptions with the db only. I feel a little strongly here becasue we create an object and do work to make the collection that will just be lost and doesn't need to be done to perform this operation meanwhile using the helper I mention still shares a majority of the code.

Reply:
From slack, the indexInformation API and the way listIndexes work will create a collection anyway. We can leave this as is